emission/net/ext_service/transit_matching/match_stops.py

```diff
@@ -67,28 +61,89 @@ def make_request_and_catch(overpass_query):
             all_results = []
     return all_results
 
-def get_public_transit_stops(min_lat, min_lon, max_lat, max_lon):
-    bbox_string = "%s,%s,%s,%s" % (min_lat, min_lon, max_lat, max_lon)
-    logging.debug("bbox_string = %s" % bbox_string)
-    overpass_public_transit_query_template = query_string
-    overpass_query = overpass_public_transit_query_template.format(bbox=bbox_string)
-    call_return = RETRY
-    retry_count = 0
-    while call_return == RETRY:
-        if retry_count > 0:
-            logging.info(f"call_return = {call_return}, retrying...")
-        call_return = make_request_and_catch(overpass_query)
-        logging.info(f"after retry, got {'RETRY' if call_return == RETRY else len(call_return)}...")
-        if call_return == RETRY:
-            retry_count = retry_count + 1
-            logging.info(f"after incrementing, retry_count = {retry_count}...")
-
-    logging.info(f"after all retries, retry_count = {retry_count}, call_return = {'RETRY' if call_return == RETRY else len(call_return)}...")
-    all_results = call_return
-
-    relations = [ad.AttrDict(r) for r in all_results if r["type"] == "relation" and r["tags"]["type"] == "route"]
+
+def load_overpass_query_template():
+    config = eac.get_config()
+    template_path = config.get("OVERPASS_QUERY_TEMPLATE_PATH")
+    
+    if template_path and os.path.exists(template_path):
+        try:
+            with open(template_path, 'r', encoding='UTF-8') as query_file:
+                logging.info("Loaded Overpass query template from %s", template_path)
+                return "".join(query_file.readlines())
+        except Exception as e:
+            logging.error("Error reading template file at %s: %s", template_path, e)
+    
+    # Fallback: use default inline template
+    logging.info("Using default inline Overpass query template")
+    return '''
+    (
+        node["highway"="bus_stop"]({bbox});
+        node["railway"="station"]({bbox});
+        node["public_transport"]({bbox});
+        way["railway"="station"]({bbox});
+        relation["route"]({bbox});
+    );
+    out body;
+    out count;
+    >;
+    '''
+
+def get_query_for_bbox(bbox):
+    bbox_string = "%s,%s,%s,%s" % bbox
+    template = load_overpass_query_template()
+    return template.format(bbox=bbox_string)
+
+
+def get_query_for_bboxes(bboxes):
+    query = '[out:json][timeout:25];\n'
```

Comment:
The hard part about making the query configurable is that the query now has 2 distinct parts: the header which is defined here, and the body which is repeated for X number of locations
The current solution only makes the body configurable. Now that I am thinking about it, I am not even sure if we need the body to be configurable because the code expects those specific features when it parses the result. If someone wants to change what features are queried, they will have to make code changes anyway

If there is one thing that definitely should be configurable, I think it is the timeout threshold which goes in the header and is currently 25